templates/addons/aurora/mysql-cf.yml

```diff
@@ -0,0 +1,171 @@
+    Default: 1000
+Resources:
+  AuroraKMSCMK:
+    Type: 'AWS::KMS::Key'
+    DeletionPolicy: Retain
+    Properties:
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            Action: 'kms:*'
+            Resource: '*'
+          - Effect: Allow
+            Principal:
+              AWS: '*'
+            Action:
+              - 'kms:Encrypt'
+              - 'kms:Decrypt'
+              - 'kms:ReEncrypt*'
+              - 'kms:GenerateDataKey*'
+              - 'kms:CreateGrant'
+              - 'kms:ListGrants'
+              - 'kms:DescribeKey'
+            Resource: '*'
+            Condition:
+              StringEquals:
+                'kms:CallerAccount': !Ref 'AWS::AccountId'
+                'kms:ViaService': !Sub 'rds.${AWS::Region}.amazonaws.com'
+
+  AuroraKMSCMKAlias:
+    Type: 'AWS::KMS::Alias'
+    DeletionPolicy: Retain
+    DependsOn: ['{{logicalIDSafe .Name}}']
+    Properties:
+      AliasName: !Sub 'alias/${ {{- logicalIDSafe .Name -}} }'
+      TargetKeyId: !Ref AuroraKMSCMK
+
+  DBSubnetGroup:
+    Type: 'AWS::RDS::DBSubnetGroup'
+    Properties:
+      DBSubnetGroupDescription: Group of Copilot private subnets for Aurora cluster.
+      SubnetIds:
+        !Split [',', { 'Fn::ImportValue': !Sub '${App}-${Env}-PrivateSubnets' }]
+
+  DBClusterSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: The Security Group for the database cluster.
+      SecurityGroupIngress:
+        - ToPort: !Ref DBPort
+          FromPort: !Ref DBPort
+          IpProtocol: tcp
+          Description: From Environment Security Group
+          SourceSecurityGroupId:
+            Fn::ImportValue:
+              !Sub '${App}-${Env}-EnvironmentSecurityGroup'
+      VpcId:
+        Fn::ImportValue:
+          !Sub '${App}-${Env}-VpcId'
+      SecurityGroupEgress:
+        - IpProtocol: '-1'
+          Description: Allow all outbound traffic by default
+          CidrIp: 0.0.0.0/0
+
+  AuroraSecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub '/${App}/${Env}/${Name}'
+      Description: !Sub Aurora main user secret for ${AWS::StackName}
+      GenerateSecretString:
+        SecretStringTemplate: '{"username": "admin"}'
+        GenerateStringKey: "password"
+        ExcludePunctuation: true
+        IncludeSpace: false
+        PasswordLength: 16
+
+  {{- if .ParameterGroup}}
+
+  #   DBClusterParameterGroup:
+  #     Type: 'AWS::RDS::DBClusterParameterGroup'
+  #     Properties:
+  #       Description: !Ref 'AWS::StackName'
+  #       Family: 'aurora-mysql5.7'
+  #       Parameters:
+  #         client_encoding: 'UTF8'
+  {{- else}}
+
+  DBClusterParameterGroup:
+    Type: 'AWS::RDS::DBClusterParameterGroup'
+    Properties:
+      Description: !Ref 'AWS::StackName'
+      Family: 'aurora-mysql5.7'
```

Comment:
We will need to change the family based on the engine version, I believe. 

For MySQL 5.6, we can use `aurora5.6`
For MySQL 5.7, `aurora-mysql5.7`
For Postgres, `aurora-postgresql10`

Reply:
The parameters will also need to change per engine:
For MySQL engines, we should use the parameter 
`character_set_database: utf32` instead of `client_encoding:utf8` as client encoding will fail to deploy with MySQL.